File: server/src/main/java/gestion_conges/server/entities/Absence.java
package gestion_conges.server.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Getter @Setter @Accessors(chain = true)
@NoArgsConstructor @AllArgsConstructor
public class Absence
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Column(nullable = false)
    private LocalDate dateDebut;
    @Column(nullable = false)
    private LocalDate dateFin;
    private String motif;
    @ManyToOne
    private TypeAbsence type;
    @ManyToOne
    private StatutAbsence statut;
}

File: client/src/app/models/absence.ts
import { StatutAbsence } from "./statut-absence";
import { TypeAbsence } from "./type-absence";

export interface Absence
{
  id: number;
  dateDebut: Date;
  dateFin: Date;
  motif: string;
  type: TypeAbsence;
  statut: StatutAbsence;
}
